pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -74,16 +76,56 @@ type cloudWatchExecutor struct {
 
 func (e *cloudWatchExecutor) newSession(region string) (*session.Session, error) {
 	dsInfo := e.getDSInfo(region)
-	creds, err := getCredentials(dsInfo)
+	regionCfg := &aws.Config{Region: aws.String(dsInfo.Region)}
+	cfgs := []*aws.Config{
+		regionCfg,
+	}
+	// Choose authentication scheme based on the type chosen for the data source
+	// Basically, we support the following methods:
+	// Shared credentials: Providing access key pair sourced from user's AWS credentials file
+	// Static credentials: Providing access key pair directly
+	// SDK: Leave it to SDK to decide
+	switch dsInfo.AuthType {
+	case "credentials":
+		plog.Debug("Authenticating towards AWS with shared credentials", "profile", dsInfo.Profile,
+			"region", dsInfo.Region)
+		cfgs = append(cfgs, &aws.Config{
+			Credentials: credentials.NewSharedCredentials("", dsInfo.Profile),
+		})
+	case "keys":
+		plog.Debug("Authenticating towards AWS with an access key pair", "region", dsInfo.Region)
+		cfgs = append(cfgs, &aws.Config{
+			Credentials: credentials.NewStaticCredentials(dsInfo.AccessKey, dsInfo.SecretKey, ""),
+		})
+	case "sdk":
+		plog.Debug("Authenticating towards AWS with default SDK method", "region", dsInfo.Region)
+	default:
+		return nil, fmt.Errorf(`%q is not a valid authentication type - expected "credentials", "keys" or "sdk"`,
+			dsInfo.AuthType)
+	}
+	sess, err := newSession(cfgs...)
```

Comment:
Is this call expensive and/or what's the impact when `dsInfo.AssumeRoleArn != ""`?  If any, move this to else-clause of `dsInfo.AssumeRoleArn != ""`` below.

Reply:
@marefr I don't think that will work, since this `sess` variable is used within `if dsInfo.AssumeRoleArn != ""`.

Reply:
You have to define the `sess` above, like `var sess *session.Session` then it will work. No?

Reply:
@marefr Ah, will check when I have the time.

Reply:
@marefr Are you aware that `sess` created here is used also on line 131, in re-assigning to `sess`?

Reply:
No I didn't understand that :+1: